specs/CriticalRestartRequired.md

```diff
@@ -0,0 +1,105 @@
+Critical Update Available
+===
+
+# Background
+As WebView2 developers, we often have to author ECS (experimentation and configuration service - 
+used to perform A/B testing or remotely disable features) configurations to toggle feature flags. 
+However, once these payloads are received, there is no way to restart the WebView2 and apply the
+payload. The purpose of this API is to detect such critical payloads and inform the end developer
+so they may restart their app or their WebView2 or other appropriate action.
+
+# Examples
+## WinRT and .NET   
+```c#
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    if (e.IsSuccess)
+    {
+        // ...
+        webView.CoreWebView2.Environment.CriticalRestartRequired += WebView_CriticalRestartRequired;
+    }
+}
+
+void WebView_CriticalRestartRequired(object sender, object e)
+{
+    // Depending on your app experience, you may or may not
+    // want to prompt user to restart the app.
+    RestartIfSelectedByUser();
+}
+```
+
+## Win32 C++
+```cpp
+void CoreWebView2InitializationCompleted() {
+    wil::com_ptr<ICoreWebView2Environment> m_webViewEnvironment;
+    auto env10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
```

Comment:
You forgot to check whether the `try_query` succeeded. C++ samples are version-agile, so there should be a
```cpp
if (env10)
{
...
}
```
guard around the code that uses `env10`.

As written, if ICoreWebView2Environment10 is not supported, it crashes on a null pointer, and we lose the reason why the QI failed. If you want to crash if ICoreWebView2Environment10 is not available, then change to `.query<>` so the QI failure reason is in the Watson report. [If there is no difference between two options, choose the one that is easier to debug](https://devblogs.microsoft.com/oldnewthing/20170725-00/?p=96676).

Reply:
We don't want the crash. Please fix.